base/reflection.jl

```diff
@@ -1628,21 +1628,24 @@ function code_typed_by_type(@nospecialize(tt::Type);
     return asts
 end
 
-function code_typed_opaque_closure(@nospecialize(oc::Core.OpaqueClosure);
-                                   debuginfo::Symbol=:default, _...)
+function get_oc_code_rt(@nospecialize(oc::Core.OpaqueClosure))
     ccall(:jl_is_in_pure_context, Bool, ()) && error("code reflection cannot be used from generated functions")
     m = oc.source
     if isa(m, Method)
         code = _uncompressed_ir(m, m.source)
-        debuginfo === :none && remove_linenums!(code)
-        # intersect the declared return type and the inferred return type (if available)
-        rt = typeintersect(code.rettype, typeof(oc).parameters[2])
-        return Any[code => rt]
+        return Pair{CodeInfo,Any}(code, typeof(oc).parameters[2])
     else
         error("encountered invalid Core.OpaqueClosure object")
     end
 end
 
+function code_typed_opaque_closure(@nospecialize(oc::Core.OpaqueClosure);
+                                   debuginfo::Symbol=:default, _...)
+    (code, rt) = get_oc_code_rt(oc)
+    debuginfo === :none && remove_linenums!(code)
+    return Any[code=>rt]
```

Comment:
Was Shuhei indicating this should be a Pair also?
```suggestion
    return Any[Pair{CodeInfo,Any}(code, rt)]
```